src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -49,6 +49,10 @@ private bool UpdateContext(SessionId sessionId, ExecutionContext executionContex
 
         protected override async Task<bool> AcceptEvent(SessionId sessionId, string method, JObject args, CancellationToken token)
         {
+            // Don't process events from sessions we aren't tracking
+            if (!contexts.TryGetValue(sessionId, out ExecutionContext context))
```

Comment:
You can't move this here, it will end up skipping the events that bind the context.  You can split the switch into two parts, those that require a context and those that don't.  As a warning you need to be very careful with the context binding logic to keep it working in the attach case as well.  We don't have good tests for that and it is easy to break.